airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java

```diff
@@ -338,6 +349,54 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(final
       LOGGER.info("Using PK + CDC");
       return MySqlInitialReadUtil.getCdcReadIterators(database, catalog, tableNameToTable, stateManager, emittedAt, getQuoteString());
     } else {
+      if (isAnyStreamIncrementalSyncMode(catalog)) {
+        final MySqlCursorBasedStateManager cursorBasedStateManager = new MySqlCursorBasedStateManager(stateManager.getRawStateMessages(), catalog);
```

Comment:
Can we move this to `MySqlInitialReadUtil`? I assume this is the code that is doing the main logic of figuring out which streams to sync via pk?

Reply:
Moved it! I actually eliminated the filtering based on state_type = cursor_based since anything that's not PK is implicitly cursor_based.